src/hazelcore/semantics/CursorInfo.re

```diff
@@ -310,10 +338,11 @@ let rec cursor_info_typ =
 
 let rec _ana_cursor_found_pat =
         (ctx: Contexts.t, p: UHPat.t, ty: HTyp.t)
-        : option((typed, node, Contexts.t)) =>
+        : option((typed, uses_list => node, Contexts.t, option(Var.t))) =>
```

Comment:
This should return a deferrable_t, or something similar in structure to it, rather than always returning a deferred node.

You might want to change deferrable_t to be a parameterized type rather than fixing it to return a `t` in the non-deferred case and a `uses_list -> t` in the deferred case, so you can use it in situations like this?

Reply:
That's a great idea! I would change `deferrable_t` to `deferrable('t)`.